src/garage/trainer.py

```diff
@@ -155,6 +159,20 @@ def __init__(self, snapshot_config):
         self._worker_class = None
         self._worker_args = None
 
+        # only used for TensorFlow algorithms
+        self._tf = tf
```

Comment:
rather than having a boolean, can you just make a session if you find tensorflow in sys.modules?